build/scripts/PullRequestValidation/LinkPullRequestToWorkItem.ps1

```diff
@@ -0,0 +1,52 @@
+using module ..\GitHub\GitHubPullRequest.class.psm1
+using module ..\GitHub\GitHubIssue.class.psm1
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string] $PullRequestNumber,
+    [Parameter(Mandatory = $true)]
+    [string] $Repository
+)
+
+function Update-GitHubPullRequest() {
+    param(
+        [Parameter(Mandatory = $false)]
+        [string] $Repository,
+        [Parameter(Mandatory = $false)]
+        [object] $PullRequest,
+        [Parameter(Mandatory = $false)]
+        [string[]] $IssueIds
+    )
+
+    $pullRequestBody = $PullRequest.PullRequest.body
+
+    # Find all ADO workitems linked to the pull request
+    foreach ($issueId in $IssueIds) {
+        Write-Host "Trying to link workitems from $issueId to pull request $PullRequestNumber"
+        $issue = [GitHubIssue]::Get($issueId, $Repository)
+        $adoWorkItems = $issue.GetLinkedADOWorkitems()
+        if (-not $adoWorkItems) {
+            Write-Host "No ADO workitems found in issue $issueId"
+            continue
+        }
+
+        foreach ($adoWorkItem in $adoWorkItems) {
+            if ($pullRequestBody -notmatch "AB#$($adoWorkItem)") {

```

Comment:
Similar to `GetLinkWorkItemIDs` in the PR class, can we have a `IsLinkedToWorkItem($workItemID)`?

Or `LinkToWorkItem($workItemID)` that links the PR to the work item (if not already linked). E.g. adds "Fixes AB#$workItemID" to the PR description?

My idea is to have the "linking logic" (the AB# syntax) hidden in the classes.